elastic/kibana
# [Synthetics] add params to project, inline, and recorder monitors

## Summary

Resolves: https://github.com/elastic/kibana/issues/135594

Adds parameters to inline and recorder monitors. Likewise, also visualizes parameters for project monitors under Advanced Browser Options

Project monitors (read only)
<img width="1064" alt="Screen Shot 2022-07-19 at 11 16 27 AM" src="https://user-images.githubusercontent.com/11356435/179789183-f12ab3ff-fee4-4b67-999c-3e4d855dead2.png">

Inline monitors
<img width="539" alt="Screen Shot 2022-07-19 at 11 24 24 AM" src="https://user-images.githubusercontent.com/11356435/179789120-ae3a061e-41da-427a-830f-bde7e65cc76b.png">


Recorder
<img width="535" alt="Screen Shot 2022-07-19 at 11 24 38 AM" src="https://user-images.githubusercontent.com/11356435/179789160-0cb98cc4-8b95-45b4-940f-2a26dc97a712.png">

Testing
--
1. Navigate to monitor management. Ensure that your local set up is able to received indexed document, either by using the synthetics service locally or by connecting to an oblt-cli cluster.
2. Add a browser monitor with an inline script that references a param. For example, you could use 
```
  step('launch app', async () => {
    await page.goto(params.url);
  });
```
3. Add the corresponding parameters, for example you could use
```
{
   "url": "https://google.com
}
```
4. Save your monitor. 
5. Navigate to Uptime. Ensure that document referenced the required param appropriately
6. Navigate back to monitor management. Add a browser monitor with a recorder script. Make sure you edit your recorder script to reference a param
7. Add the corresponding param
8. Navigate to Uptime. Ensure the document referenced the required param appropriately.
9. Navigate to Monitor management. Generate an api key for project monitors.
10. Push a project monitor with a params option.
11. Navigate to Monitor management and click edit on the project monitor. Click browser advanced settings. Ensure the params are correct.
